### PR TITLE
Add & remove

### PR DESCRIPTION
**Pull request by CarlosZ96**
An edit function is created to enable editing of the input with the task description and save this data in the list of objects, pressing enter changes the value and disables editing again until it is clicked.

### CODE REVIEW

File: src/task.js [hunk 1]
```diff
@@ -1,27 +1,85 @@
...
+    });
+  }
+
+  EditTask(id) {
+    const index = this.tasks.findIndex((task) => task.id === Number(id));
+    if (index !== -1) {
+      const newDescription = document.getElementById('TaskInput').value;
+      this.tasks[index].description = newDescription;
+      localStorage.setItem('tasks', JSON.stringify(this.tasks));
+    }
```
Comment (ridwanediallo) [inline src/task.js]: - [ ] Hello there, @CarlosZ96. You did an excellent job bringing your project to this point. However, there is a problem when we attempt to edit a task. When we click on a task to edit it, it is removed from local storage. Please ensure that this problem is resolved and that we are able to successfully edit a task. Not only will you meet the requirement, but your app will also provide a better user experience. :ok_hand: 😊

### Demo :point_down: 
[screen-capture.webm](https://github.com/CarlosZ96/Tasks/assets/89228743/8120e5fa-d6f2-41f6-82e5-d99ac4009939)

**Review by ridwanediallo — CHANGES_REQUESTED** [review]
# Changes Required ♻️
## Hi @CarlosZ96 , 
AlmostThereReachGIF

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used GitHub flow correctly. ✔
- Your app is free of linters errors. ✔


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/task.js [hunk 2]
```diff
@@ -1,27 +1,100 @@
...
+      listcontainer.appendChild(taskcontainer);
+      const btnrevome = taskcontainer.querySelector('.removeBtn');
+      btnrevome.addEventListener('click', (event) => {
+        event.preventDefault();
+        const { id } = event.target.dataset;
+        this.DeleteTask(id);
+        this.ShowTask();
+        for (let i = 0; i < task.length; i + 1) {
+          task[i].id = (i + 1);
+        }
```
Comment (Olamarx) [inline src/task.js:47]: - [OPTIONAL] I would recommend you work on the id, it does not rearrange after every delete and it is a requirement but the previous reviewer missed it that is why I made it optional.

**Review by Olamarx — APPROVED** [review]
Hi @CarlosZ96, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- Linters passed ✔️
- Readme file professional ✔️
- All methods in place ✔️

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**